source/common/tcp_proxy/tcp_proxy.cc

```diff
@@ -391,6 +391,8 @@ Network::FilterStatus Filter::initializeUpstreamConnection() {
   route_ = pickRoute();
 
   const std::string& cluster_name = route_ ? route_->clusterName() : EMPTY_STRING;
+  getStreamInfo().setUpstreamEndpointInfo(route_);
+  // getStreamInfo().setRouteEntry(std::dynamic_pointer_cast<const Router::RouteEntry>(route_));
```

Comment:
Left over comment?

Reply:
yes.. will remove.. I had hoped this would work but it obviously doesn't ..
and because of which we have to add 2 new functions for UpstreamEndpointInfo..